### PR TITLE
docs: write v6 upgrade guide

### PR DESCRIPTION
**Pull request by Justintime50**
# Description

Adds the v6 upgrade guide in preperation for the v6 release

# Testing

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

File: UPGRADE_GUIDE.md [hunk 1]
```diff
@@ -2,9 +2,150 @@
...
+
+New EasyPostError (extends the builtin Error):
+
+- `errors/general/filtering_error`
+- `errors/general/invalid_object_error`
+- `errors/general/invalid_parameter_error`
+- `errors/general/missing_parameter_error`
+- `errors/general/signature_verification_error`
+
+ApiErrors will behave like the previous Error class did. They will include a `message`, `errors`, `code`, and `statusCode`. This class extends the builtin Error which only contains a message, used for exceptions thrown directly from this library.
```
Comment (jchen293) [inline UPGRADE_GUIDE.md:123]:
```suggestion
ApiErrors will behave like the previous Error class did. They will include a `message`, `errors`, `code`, and `statusCode`. This class extends the EasyPostError which only contains a message, used for exceptions thrown directly from this library.
```

**Review by jchen293 — DISMISSED** [review]

File: UPGRADE_GUIDE.md (hunk 1 above)
Comment (Justintime50) [reply UPGRADE_GUIDE.md]: Nice catch

**Review by jchen293 — APPROVED** [review]

File: UPGRADE_GUIDE.md [hunk 2]
```diff
@@ -2,9 +2,150 @@
...
+
+- `errors/api/api_error`
+- `errors/api/encoding_error`
+- `errors/api/external_api_error`
+- `errors/api/forbidden_error`
+- `errors/api/gateway_timeout_error`
+- `errors/api/http_error`
+- `errors/api/internal_server_error`
+- `errors/api/invalid_request_error`
+- `errors/api/json_error`
```
Comment (nwithan8) [inline UPGRADE_GUIDE.md:104]: Is JSON error an API error or a local error? Writing the docstring now, and it's not used anywhere. In C#, this error is for when de/serialization fails locally, but not sure if it's the same (or even needed) in Node.

**Review by nwithan8 — CHANGES_REQUESTED** [review]

File: UPGRADE_GUIDE.md (hunk 2 above)
Comment (jchen293) [reply UPGRADE_GUIDE.md:104]: Probably not in Node, I should have excluded this for Node

Comment (nwithan8) [reply UPGRADE_GUIDE.md:104]: I'll delete it in my docstrings PR

Comment (Justintime50) [reply UPGRADE_GUIDE.md:104]: Ok, so wanna stamp this one? 😉

**Review by nwithan8 — APPROVED** [review]

**Pull request merged**